packages/fx-core/src/component/resource/botService/botService.ts

```diff
@@ -124,7 +124,7 @@ export class BotService extends AzureResource {
       teamsBotState.botPassword = regRes.value.botPassword;
       return ok(undefined);
     } catch (e) {
-      if (e.name == ErrorNames.CREATE_BOT_REGISTRATION_API_ERROR && hasBotIdInEnvBefore) {
+      if (e.innerError.teamsfxUrlName == "<create-bot-registration>" && hasBotIdInEnvBefore) {
```

Comment:
thought this may be undefined. innerError?.teamsfxUrlName

Reply:
Good catch! It's better to use `?` although the IDE didn't propose the same suggestion.